src/pages/patterns/tables/basic-formatting.mdx

```diff
@@ -0,0 +1,441 @@
+
+![Pagination Cursor](figma:patterns-tables-basic-formatting-pagination-cursor.png)
+
+##### Offset pagination
+
+Use [Offset pagination](/components/pagination#offset-pagination) when you know the number of records or where you are
+in the data set.
+
+- Ensure the margin between the table component and the pagination is 20px
+
+![Pagination Offset](figma:patterns-tables-basic-formatting-pagination-offset.png)
+
+##### Select all for pagination
+
+In paginated tables, you can select all checkboxes but that selection only affects the current page being shown.
+
+- Users should only be able to take action on what they can see
+
+![Pagination Selecting All](figma:patterns-tables-basic-formatting-pagination-selecting-all.png)
+
+#### Infinite scrolling
+
+In cases where the database supports infinite scrolling, there is no need to say how many pages of data there are.
+
+If you have an infinitely scrolling table, you should also offer [sorting](/patterns/tables/basic-formatting#sorting) to
+help users navigate the data.
+
+##### Select all for infinite scrolling
+
+In infinitely scrolling tables, you can only manually select items.
+
+##### Vertical scrolling
+
+For tables with more rows than can be displayed in the viewport, the header row is fixed at the top of the viewport.
+Keeping the column headers within view while scrolling helps users maintain context.
+
+import TableVerticalScrolling from '../examples/tables/TableVerticalScrolling.tsx';
+import TableVerticalScrollingCode from '!!raw-loader!../examples/tables/TableVerticalScrolling.tsx';
+
+<CodeExample code={TableVerticalScrollingCode}>
+  <TableVerticalScrolling />
+</CodeExample>
+
+##### Virtualization
+
+Virtualization means what you see in the browser is what is in the DOM. When something scrolls out of view, it gets
+removed from the page DOM. When something scrolls into view, it is added to the DOM.
+
+- Use [react-window](https://github.com/bvaughn/react-window) to efficiently render large data sets
+
+import TableVirtualized from '../examples/tables/TableVirtualized.tsx';
+import TableVirtualizedCode from '!!raw-loader!../examples/tables/TableVirtualized.tsx';
+
+<CodeExample code={TableVirtualizedCode}>
+  <TableVirtualized />
+</CodeExample>
+
+#### Sorting
+
+Sorting is implemented by selecting arrows in table headers.
+
+- Clicking the header text will sort by ascending order (A→Z, 1→9, Date and Time, Priority (High→Low))
+- Clicking again will reverse the order
+- Clicking a third time will remove sorting
+
+If any ordering applied to a table including default ordering, the sorting icon should be present. This indicates how
+the table is being ordered and also creates a better affordance for the control.
+
+import TableSort from '../examples/tables/TableSort.tsx';
+import TableSortCode from '!!raw-loader!../examples/tables/TableSort.tsx';
+
+<CodeExample code={TableSortCode}>
+  <TableSort />
+</CodeExample>
+
+#### Draggable
+
```

Comment:
I am trying to review this pattern holistically from an accessibility standpoint here:

Is the accessibility support from react-beautiful-dnd enough for documenting how to do accessibility for Garden Table row reordering? I ran through the screen reader and it said "you have moved item from position 1 to position 2". Any chance there's a way to recommend how to change "item" to "Raspberry" or an identifying row name for developers to follow? What do you think @jgorfine-zendesk?

Reply:
Hey @mtomcal — I totally agree with you: it might be more helpful if the label were more specific, i.e., had the name of the "item" in it. 

I wonder if it would also be helpful to include the total number of positions. Looking at the example from the React Beautiful DnD guidelines:
```
You have moved the task to position 2
``` 

it could be modified to be like this: 
```
You have moved the task “Raspberry“ to position 2 of 5
```

In the context of a table, that might sound like this:
```
You have moved [string value of row's <th> element] to row 2 of 5
```

Reply:
Could these potential changes be reserved for a separate unit of work since this example currently exists [in production](https://garden.zendesk.com/components/table#draggable)? I'm happy to create a task for it if it's necessary.

Reply:
Definitely task for this as a tech debt item for a11y since this predates this effort. 👍